golang/cosmos/app/upgrade.go

```diff
@@ -131,29 +131,37 @@ func getVariantFromUpgradeName(upgradeName string) string {
 		return "DEVNET"
 	case "UNRELEASED_emerynet":
 		return "EMERYNET"
-		// Noupgrade for this version.
 	case "UNRELEASED_BASIC":
+		// Noupgrade for this version.
 		return ""
 	default:
 		return ""
 	}
 }
 
-// func upgradeMintHolderCoreProposal(upgradeName string) (vm.CoreProposalStep, error) {
-// 	variant := getVariantFromUpgradeName(upgradeName)
+func upgradeMintHolderCoreProposal(targetUpgrade string) (vm.CoreProposalStep, error) {
+	return buildProposalStepFromScript(targetUpgrade, "@agoric/builders/scripts/vats/upgrade-mintHolder.js")
+}
 
-// 	if variant == "" {
-// 		return nil, nil
-// 	}
+func restartFeeDistributorCoreProposal(targetUpgrade string) (vm.CoreProposalStep, error) {
+	return buildProposalStepFromScript(targetUpgrade, "@agoric/builders/scripts/inter-protocol/replace-feeDistributor-combo.js")
+}
 
-// 	return buildProposalStepWithArgs(
-// 		"@agoric/builders/scripts/vats/upgrade-mintHolder.js",
-// 		"defaultProposalBuilder",
-// 		map[string]any{
-// 			"variant": variant,
-// 		},
-// 	)
-// }
+func buildProposalStepFromScript(targetUpgrade string, builderScript string) (vm.CoreProposalStep, error) {
+	variant := getVariantFromUpgradeName(targetUpgrade)
+
+	if variant == "" {
+		return nil, nil
+	}
+
+	return buildProposalStepWithArgs(
+		builderScript,
+		"defaultProposalBuilder",
+		map[string]any{
+			"variant": variant,
```

Comment:
Really unfortunate we still have the need for variants in this upgrade. We need to find a way out of this pickle, but I'm not gonna try to figure out why this upgrade needs special casing per network now.

In the mean time, we can hopefully prioritize https://github.com/Agoric/agoric-sdk/issues/10837

Reply:
I've read #10837 and don't understand the path forward.

This distinction is because we want to be able to validate the feeDistributor, and waiting a full hour-long cycle for it seems infeasible. We have other timing-related variations for auctions. In order to get rid of variant configurations for test, we need another solution.

Reply:
I think my core question was why these config have to be provided as core proposal argument, and cannot be stored in the config of the contract we're upgrading. That's what https://github.com/Agoric/agoric-sdk/issues/10956 asks. We can punt on it for now.